extensions/s3/src/main/java/io/deephaven/extensions/s3/S3Instructions.java

```diff
@@ -135,6 +144,48 @@ public int numConcurrentWriteParts() {
         return DEFAULT_NUM_CONCURRENT_WRITE_PARTS;
     }
 
+    /**
+     * The default profile name used for configuring the default region, credentials, etc., when reading or writing to
+     * S3. If not provided, the AWS SDK picks the profile name from the 'aws.profile' system property, the "AWS_PROFILE"
+     * environment variable, or defaults to "default".
+     *
+     * @see ClientOverrideConfiguration.Builder#defaultProfileName(String)
+     */
+    public abstract Optional<String> profileName();
```

Comment:
We should probably add a comment in all of these javadocs: "This assumes that the credentials are provided via the profile; if that is not the case, you must explicitly set credentials".